src/metabase/search/scoring.clj

```diff
@@ -198,5 +206,19 @@
   "Returns a map with the `:score` and `:result`—or nil. The score is a vector of comparable things in priority order."
   [query-string result]
   (when-let [hit (text-score-with-match query-string result)]
-    {:score (combined-score hit)
-     :result (serialize hit)}))
+    (let [score (combined-score hit)]
+      {:score score
+       :result (serialize hit score)})))
+
+(defn top-results
+  "Given a reducible collection (i.e., from `jdbc/reducible-query`) and a transforming function for it, applies the
+  transformation and returns a seq of the results sorted by score. The transforming function is expected to output
+  maps with `:score` and `:result` keys."
+  [reducible-results xf]
+  (->> reducible-results
+       (transduce xf accumulate-top-results)
+       ;; The accumulator doesn't guarantee order
+       (sort-by :score)
```

Comment:
i think you can drop this `(sort-by :score)`. the accumulator does guarantee order. It looks on `.poll`

> The head of this queue is the least element with respect to the specified ordering. If multiple elements are tied for least value, the head is one of those elements -- ties are broken arbitrarily. The queue retrieval operations poll, remove, peek, and element access the element at the head of the queue. 

https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/util/PriorityQueue.html

Reply:
Good call, done. I was mistakenly looking at the `toArray` documentation